src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerExtensions.cs

```diff
@@ -95,7 +99,22 @@ public static IApplicationBuilder UseExceptionHandler(this IApplicationBuilder a
                 throw new ArgumentNullException(nameof(options));
             }
 
-            return app.UseMiddleware<ExceptionHandlerMiddleware>(Options.Create(options));
+            return app.Use(next =>
```

Comment:
All middlewares that modifies paths will now need to follow this pattern to work for Minimal APIs. I don't think this is very intuitive though.

Also, a similar fix is likely needed for RewriteMiddleware

Reply:
This also needs to be more specific, we probably need to distinguish between minimal API vs regular apps to figure out if we need to handle the middleware differently.